engine/orchestration/src/main/java/com/cloud/vm/VirtualMachineManagerImpl.java

```diff
@@ -1058,10 +1061,7 @@ public void orchestrateStart(final String vmUuid, final Map<VirtualMachineProfil
 
         DataCenterDeployment plan = new DataCenterDeployment(vm.getDataCenterId(), vm.getPodIdToDeployIn(), null, null, null, null, ctx);
         if (planToDeploy != null && planToDeploy.getDataCenterId() != 0) {
-            if (s_logger.isDebugEnabled()) {
-                s_logger.debug("advanceStart: DeploymentPlan is provided, using dcId:" + planToDeploy.getDataCenterId() + ", podId: " + planToDeploy.getPodId() +
-                        ", clusterId: " + planToDeploy.getClusterId() + ", hostId: " + planToDeploy.getHostId() + ", poolId: " + planToDeploy.getPoolId());
-            }
+            s_logger.debug(DeploymentPlanningManagerImpl.logDeploymentWithoutException((VirtualMachine) vm, planToDeploy, planToDeploy.getAvoids(), (DeploymentPlanner) plan));
```

Comment:
While testing, I found that this line causes a `java.lang.ClassCastException: class com.cloud.deploy.DataCenterDeployment cannot be cast to class com.cloud.deploy.DeploymentPlanner`.

Reply:
Done, thanks.